src/zenml/config/step_configurations.py

```diff
@@ -44,6 +44,7 @@ class PartialArtifactConfiguration(StrictBaseModel):
     """Class representing a partial input/output artifact configuration."""
 
     materializer_source: Optional[Tuple[Source, ...]] = None
+    default_materializer_source: Optional[Source] = None
```

Comment:
If the type annotation is Any, we now set the `default_materializer_source` to the default materializer of the registry (usually cloudpickle, but can be overwritten by users). Now when executing a step and it returns an `int`, we do the following:
- Check if an explicit materializer should be used (this is the case if the step has a type annotation or a materializer was manually configured)
- If not, query the materializer registry to find a materializer at runtime (this will only work if a materializer was registered for the type)
- If not, use the materializer defined by `default_materializer_source`

The second step would never happen if we were to store the default materializer as part of the `materializer_source` attribute. I hope that explains it?